Submitted By: Wayne Blaszczyk
Date: 2010-06-06
Initial Package Version: 0.6.5
Origin: http://git.gnome.org/browse/gir-repository/log/
Upstream Status: commited
Description: Fixed various DBusGLib-1.0.gir issues.

diff -Naur gir-repository-0.6.5/gir/DBusGLib-1.0.gir gir-repository-0.6.5.fixed/gir/DBusGLib-1.0.gir
--- gir-repository-0.6.5/gir/DBusGLib-1.0.gir	2009-07-10 03:31:54.000000000 +1000
+++ gir-repository-0.6.5.fixed/gir/DBusGLib-1.0.gir	2010-06-06 16:44:48.000000000 +1000
@@ -2,8 +2,9 @@
 <repository version="1.0"
             xmlns="http://www.gtk.org/introspection/core/1.0"
             xmlns:c="http://www.gtk.org/introspection/c/1.0"
-	    xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
-  <include name="GObject"/>
+            xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
+  <include name="GObject" version="2.0"/>
+  <package name="dbus-glib-1"/>
   <namespace name="DBusGLib" version="1.0" shared-library="dbus-glib-1">
     <record name="Connection" c:type="DBusGConnection"/>
     <record name="MethodInvocation" c:type="DBusGMethodInvocation"/>
@@ -11,16 +12,14 @@
       glib:type-name="DBusGProxy" glib:get-type="dbus_g_proxy_get_type">
     </class>
     <record name="ProxyClass" c:type="DBusGProxyClass"/>
-
-    <function name="connection_open" c:identifier="dbus_g_connection_open"
-     throws="1">
+    <function name="connection_open" c:identifier="dbus_g_connection_open" throws="1">
       <return-value transfer-ownership="full">
         <type name="Connection" c:type="DBusGConnection*"/>
       </return-value>
       <parameters>
-        <parameter name="address">
-	  <type name="utf8" c:type="char*"/>
-	</parameter>
+        <parameter name="address" transfer-ownership="none">
+          <type name="utf8" c:type="char*"/>
+        </parameter>
       </parameters>
     </function>
   </namespace>
diff -Naur gir-repository-0.6.5/gir/Makefile.am gir-repository-0.6.5.fixed/gir/Makefile.am
--- gir-repository-0.6.5/gir/Makefile.am	2009-08-29 11:42:39.000000000 +1000
+++ gir-repository-0.6.5.fixed/gir/Makefile.am	2010-06-06 16:37:28.000000000 +1000
@@ -12,8 +12,7 @@
 endif
 EXTRA_DIST += $(AVAHI_GIRSOURCES)
 
-MISSIONCONTROL_GIRSOURCES = DBusGLib-1.0.gir \
-			    TelepathyGLib.gir \
+MISSIONCONTROL_GIRSOURCES = TelepathyGLib.gir \
 			    LibTelepathy.gir
 if BUILD_MISSIONCONTROL
 CUSTOM_GIRSOURCES += $(MISSIONCONTROL_GIRSOURCES)
@@ -29,6 +28,8 @@
 EXTRA_DIST += $(srcdir)/dbus.c $(srcdir)/dbus.h
 
 if BUILD_DBUS
+CUSTOM_GIRSOURCES += DBusGLib-1.0.gir
+
 DBus-1.0.gir: libgirepo-DBus-custom.la
 	$(INTROSPECTION_SCANNER) -v --namespace DBus --nsversion=1.0 \
 	    --add-include-path=$(srcdir) --add-include-path=. \
